![JohnSnowLabs](https://nlp.johnsnowlabs.com/assets/images/logo.png)

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/JohnSnowLabs/spark-nlp-workshop/blob/master/tutorials/Certification_Trainings/Healthcare/11.1.Healthcare_Code_Mapping.ipynb)

In [1]:
import json, os
from google.colab import files

if 'spark_jsl.json' not in os.listdir():
  license_keys = files.upload()
  os.rename(list(license_keys.keys())[0], 'spark_jsl.json')

with open('spark_jsl.json') as f:
    license_keys = json.load(f)

# Defining license key-value pairs as local variables
locals().update(license_keys)
os.environ.update(license_keys)

Saving spark_nlp_for_healthcare_spark_ocr_9848.json to spark_nlp_for_healthcare_spark_ocr_9848.json


In [2]:
# Installing pyspark and spark-nlp
! pip install --upgrade -q pyspark==3.4.1 spark-nlp==$PUBLIC_VERSION

# Installing Spark NLP Healthcare
! pip install --upgrade -q spark-nlp-jsl==$JSL_VERSION  --extra-index-url https://pypi.johnsnowlabs.com/$SECRET

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 310.8/310.8 MB 4.2 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 626.6/626.6 kB 5.7 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 538.9/538.9 kB 8.4 MB/s eta 0:00:00


In [4]:
import json
import os

import sparknlp_jsl
import sparknlp

from pyspark.sql import SparkSession
from pyspark.ml import Pipeline, PipelineModel

from sparknlp.base import *
from sparknlp.annotator import *
from sparknlp.util import *
from sparknlp.pretrained import PretrainedPipeline
from sparknlp_jsl.annotator import *

params = {"spark.driver.memory":"24G",
          "spark.kryoserializer.buffer.max":"2000M",
          "spark.driver.maxResultSize":"4000M"}

spark = sparknlp_jsl.start(license_keys['SECRET'],params=params)

print("Spark NLP Version :", sparknlp.version())
print("Spark NLP_JSL Version :", sparknlp_jsl.version())

spark

Spark NLP Version : 5.5.1
Spark NLP_JSL Version : 5.5.1


## Healthcare NLP for Data Scientists Course

If you are not familiar with the components in this notebook, you can check [Healthcare NLP for Data Scientists Udemy Course](https://www.udemy.com/course/healthcare-nlp-for-data-scientists/) and the [MOOC Notebooks](https://github.com/JohnSnowLabs/spark-nlp-workshop/tree/master/Spark_NLP_Udemy_MOOC/Healthcare_NLP) for each components.

# HEALTHCARE CODES MAPPING BY USING PRETRAINED PIPELINES


- **Chunk Mapper Pretrained Pipelines**

|index|model|
|-----:|:-----|
| 1| [icd10_icd9_mapping](https://nlp.johnsnowlabs.com/2022/09/30/icd10_icd9_mapping_en.html)   |
| 2| [icdo_snomed_mapping](https://nlp.johnsnowlabs.com/2022/06/27/icdo_snomed_mapping_en_3_0.html)   |
| 3| [icd10cm_snomed_mapping](https://nlp.johnsnowlabs.com/2022/06/27/icd10cm_snomed_mapping_en_3_0.html)   |
| 4| [rxnorm_ndc_mapping](https://nlp.johnsnowlabs.com/2022/06/27/rxnorm_ndc_mapping_en_3_0.html)   |
| 5| [rxnorm_umls_mapping](https://nlp.johnsnowlabs.com/2022/06/27/rxnorm_umls_mapping_en_3_0.html)   |
| 6| [snomed_icd10cm_mapping](https://nlp.johnsnowlabs.com/2022/06/27/snomed_icd10cm_mapping_en_3_0.html)   |
| 7| [snomed_icdo_mapping](https://nlp.johnsnowlabs.com/2022/06/27/snomed_icdo_mapping_en_3_0.html)   |
| 8| [snomed_umls_mapping](https://nlp.johnsnowlabs.com/2022/06/27/snomed_umls_mapping_en_3_0.html)   |
| 9| [icd10cm_umls_mapping](https://nlp.johnsnowlabs.com/2022/06/27/icd10cm_umls_mapping_en_3_0.html)   |
| 10| [mesh_umls_mapping](https://nlp.johnsnowlabs.com/2021/07/01/mesh_umls_mapping_en.html)   |
| 11| [rxnorm_mesh_mapping](https://nlp.johnsnowlabs.com/2021/07/01/rxnorm_mesh_mapping_en.html)   |



**You can find all these models and more [NLP Models Hub](https://nlp.johnsnowlabs.com/models?q=Chunk+Mapping&edition=Spark+NLP+for+Healthcare)**

In [ ]:
from sparknlp.pretrained import PretrainedPipeline

## ICD10 to ICD9 Code Mapping
This pretrained pipeline maps ICD10 codes to ICD9 codes without using any text data. You’ll just feed a comma or white space delimited ICD10 codes and it will return the corresponding ICD9 codes as a list. If there is no mapping, the original code is returned with no mapping.

In [ ]:
icd10_icd9_pipeline = PretrainedPipeline("icd10_icd9_mapping", "en", "clinical/models")

icd10_icd9_mapping download started this may take some time.
Approx size to download 579.7 KB
[OK!]


In [ ]:
icd10_icd9_pipeline.model.stages

[DocumentAssembler_78a9faf419f2,
 REGEX_TOKENIZER_cb2df0b8d959,
 CHUNKER-MAPPER_3ac48fe0df3d]

In [ ]:
icd10_icd9_pipeline.annotate('E669 R630 J988')

{'document': ['E669 R630 J988'],
 'icd10cm_code': ['E669', 'R630', 'J988'],
 'icd9_code': ['27800', '7830', '5198']}

| ICD10 | Details |
| ---------- | ----------------------------:|
| E669 | Obesity |
| R630 | Anorexia |
| J988 | Other specified respiratory disorders |



| ICD9 | Details |
| ---------- | ---------------------------:|
| 27800 | Obesity |
| 7830 | Anorexia |
| 5198 | Other diseases of respiratory system |

## ICD10CM to SNOMED Code Mapping

This pretrained pipeline maps ICD10CM codes to SNOMED codes without using any text data. You’ll just feed a comma or white space delimited ICD10CM codes and it will return the corresponding SNOMED codes as a list. For the time being, it supports 132K Snomed codes and will be augmented & enriched in the next releases.

In [ ]:
icd10_snomed_pipeline = PretrainedPipeline("icd10cm_snomed_mapping", "en", "clinical/models")

icd10cm_snomed_mapping download started this may take some time.
Approx size to download 1 MB
[OK!]


In [ ]:
icd10_snomed_pipeline.model.stages

[DocumentAssembler_9ae20ea07807,
 REGEX_TOKENIZER_8176a6b6e586,
 CHUNKER-MAPPER_5968de7123c1]

In [ ]:
icd10_snomed_pipeline.annotate('M8950 E119 H16269')

{'document': ['M8950 E119 H16269'],
 'icd10cm_code': ['M8950', 'E119', 'H16269'],
 'snomed_code': ['716868003', '170771004', '51264003']}

|**ICD10CM Code** | **ICD10CM Details** | **SNOMED Code** | **SNOMED Details** |
| ---------- | -----------:| ---------- | -----------:|
| M8950 |  Osteolysis, unspecified site | 716868003 | Multicentric osteolysis nodulosis arthropathy spectrum |
| E119 | Type 2 diabetes mellitus | 170771004 | Diabetic - follow-up default |
| H16269 | Vernal keratoconjunctivitis, with limbar and corneal involvement, unspecified eye | 51264003 | Limbal AND/OR corneal involvement in vernal conjunctivitis |


## SNOMED to ICD10CM Code Mapping

This pretrained pipeline maps SNOMED codes to ICD10CM codes without using any text data. You'll just feed a comma or white space delimited SNOMED codes and it will return the corresponding candidate ICD10CM codes as a list (multiple ICD10 codes for each Snomed code). For the time being, it supports 132K Snomed codes and 30K ICD10 codes and will be augmented & enriched in the next releases.

In [ ]:
snomed_icd10_pipeline = PretrainedPipeline("snomed_icd10cm_mapping","en","clinical/models")

snomed_icd10cm_mapping download started this may take some time.
Approx size to download 1.5 MB
[OK!]


In [ ]:
snomed_icd10_pipeline.model.stages

[DocumentAssembler_4d05b9c5d71e,
 REGEX_TOKENIZER_d8ccf33e2cd4,
 CHUNKER-MAPPER_fa42286a8f92]

In [ ]:
snomed_icd10_pipeline.annotate('716868003 170771004 51264003')

{'document': ['716868003 170771004 51264003'],
 'snomed_code': ['716868003', '170771004', '51264003'],
 'icd10cm_code': ['M89.50', 'E11.9', 'H16.269']}

| **SNOMED Code** | **SNOMED Details** |**ICD10CM Code** | **ICD10CM Details** |
| ---------- | -----------:| ---------- | -----------:|
| 716868003 | Multicentric osteolysis nodulosis arthropathy spectrum | M89.50 |  Osteolysis, unspecified site |
| 170771004 | Diabetic - follow-up default | E11.9 | Type 2 diabetes mellitus |
| 51264003 | Limbal AND/OR corneal involvement in vernal conjunctivitis | H16.269 | Vernal keratoconjunctivitis, with limbar and corneal involvement, unspecified eye |



## ICD-O to SNOMED Code Mapping

This pretrained pipeline maps ICD-O codes to SNOMED codes without using any text data. You’ll just feed a comma or white space delimited ICD-O codes and it will return the corresponding SNOMED codes as a list.

In [ ]:
icdo_snomed_pipeline = PretrainedPipeline("icdo_snomed_mapping", "en", "clinical/models")

icdo_snomed_mapping download started this may take some time.
Approx size to download 134.1 KB
[OK!]


In [ ]:
icdo_snomed_pipeline.model.stages

[DocumentAssembler_666bb5cbda3a,
 REGEX_TOKENIZER_9a9b8335227b,
 CHUNKER-MAPPER_e9822ba7753b]

In [ ]:
icdo_snomed_pipeline.annotate('8172/3 C77.5 8982/0')

{'document': ['8172/3 C77.5 8982/0'],
 'icdo_code': ['8172/3', 'C77.5', '8982/0'],
 'snomed_code': ['128646008', '5394000', '69291002']}

|**ICDO Code** | **ICDOCM Details** | **SNOMED Code** | **SNOMED Details** |
| ---------- | -----------:| ---------- | -----------:|
| 8172/3 |  Hepatocellular carcinoma, scirrhous | 128646008 | Hepatocellular carcinoma, scirrhous |
| C77.5 | Pelvic lymph nodes | 5394000 | Structure of uterine paracervical lymph node |
| 8982/0 | Myoepithelioma, NOS | 69291002 | Myoepithelial adenoma |


## SNOMED to ICD-O Code Mapping

This pretrained pipeline maps SNOMED codes to ICD-O codes without using any text data. You’ll just feed a comma or white space delimited SNOMED codes and it will return the corresponding ICD-O codes as a list.

In [ ]:
snomed_icdo_pipeline = PretrainedPipeline("snomed_icdo_mapping", "en", "clinical/models")

snomed_icdo_mapping download started this may take some time.
Approx size to download 207.8 KB
[OK!]


In [ ]:
snomed_icdo_pipeline.model.stages

[DocumentAssembler_82ca3d0764d1,
 REGEX_TOKENIZER_42013a5c2e7a,
 CHUNKER-MAPPER_2e80f745c696]

In [ ]:
snomed_icdo_pipeline.annotate('128646008 5394000 69291002')

{'document': ['128646008 5394000 69291002'],
 'snomed_code': ['128646008', '5394000', '69291002'],
 'icdo_code': ['8172/3', 'C77.5', '8982/0']}

|**SNOMED Code** | **SNOMED Details** |**ICDO Code** | **ICDOCM Details** |
| ---------- | -----------:| ---------- | -----------:|
|128646008 | Hepatocellular carcinoma, scirrhous | 8172/3 |  Hepatocellular carcinoma, scirrhous |
|5394000 | Structure of uterine paracervical lymph node | C77.5 | Pelvic lymph nodes |
|69291002 | Myoepithelial adenoma | 8982/0 | Myoepithelioma, NOS |





## ICD10CM to UMLS Code Mapping

This pretrained pipeline maps ICD10CM codes to UMLS codes without using any text data. You’ll just feed white space delimited ICD10CM codes and it will return the corresponding UMLS codes as a list. If there is no mapping, the original code is returned with no mapping.

In [5]:
icd10_umls_pipeline = PretrainedPipeline( "icd10cm_umls_mapping","en","clinical/models")

icd10cm_umls_mapping download started this may take some time.
Approx size to download 934.1 KB
[OK!]


In [8]:
import os
os.getcwd()

'/content'

In [11]:
import pickle
desc_list = pickle.load(open("/content/desc_list.pkl", "rb"))
my_string = ' '.join(desc_list)
print(my_string)

ICD-10-CM 1 A00-A09 A00 A00.0 A00.1 A00.9 A01 A01.0 A01.00 A01.01 A01.02 A01.03 A01.04 A01.05 A01.09 A01.1 A01.2 A01.3 A01.4 A02 A02.0 A02.1 A02.2 A02.20 A02.21 A02.22 A02.23 A02.24 A02.25 A02.29 A02.8 A02.9 A03 A03.0 A03.1 A03.2 A03.3 A03.8 A03.9 A04 A04.0 A04.1 A04.2 A04.3 A04.4 A04.5 A04.6 A04.7 A04.71 A04.72 A04.8 A04.9 A05 A05.0 A05.1 A05.2 A05.3 A05.4 A05.5 A05.8 A05.9 A06 A06.0 A06.1 A06.2 A06.3 A06.4 A06.5 A06.6 A06.7 A06.8 A06.81 A06.82 A06.89 A06.9 A07 A07.0 A07.1 A07.2 A07.3 A07.4 A07.8 A07.9 A08 A08.0 A08.1 A08.11 A08.19 A08.2 A08.3 A08.31 A08.32 A08.39 A08.4 A08.8 A09 A15-A19 A15 A15.0 A15.4 A15.5 A15.6 A15.7 A15.8 A15.9 A17 A17.0 A17.1 A17.8 A17.81 A17.82 A17.83 A17.89 A17.9 A18 A18.0 A18.01 A18.02 A18.03 A18.09 A18.1 A18.10 A18.11 A18.12 A18.13 A18.14 A18.15 A18.16 A18.17 A18.18 A18.2 A18.3 A18.31 A18.32 A18.39 A18.4 A18.5 A18.50 A18.51 A18.52 A18.53 A18.54 A18.59 A18.6 A18.7 A18.8 A18.81 A18.82 A18.83 A18.84 A18.85 A18.89 A19 A19.0 A19.1 A19.2 A19.8 A19.9 A20-A28 A20 A2

In [13]:
import pickle
desc_list_clean = pickle.load(open("/content/desc_list_clean.pkl", "rb"))
my_string_clean = ' '.join(desc_list_clean)
print(my_string_clean)

ICD-10-CM 1 A00-A09 A00 A000 A001 A009 A01 A010 A0100 A0101 A0102 A0103 A0104 A0105 A0109 A011 A012 A013 A014 A02 A020 A021 A022 A0220 A0221 A0222 A0223 A0224 A0225 A0229 A028 A029 A03 A030 A031 A032 A033 A038 A039 A04 A040 A041 A042 A043 A044 A045 A046 A047 A0471 A0472 A048 A049 A05 A050 A051 A052 A053 A054 A055 A058 A059 A06 A060 A061 A062 A063 A064 A065 A066 A067 A068 A0681 A0682 A0689 A069 A07 A070 A071 A072 A073 A074 A078 A079 A08 A080 A081 A0811 A0819 A082 A083 A0831 A0832 A0839 A084 A088 A09 A15-A19 A15 A150 A154 A155 A156 A157 A158 A159 A17 A170 A171 A178 A1781 A1782 A1783 A1789 A179 A18 A180 A1801 A1802 A1803 A1809 A181 A1810 A1811 A1812 A1813 A1814 A1815 A1816 A1817 A1818 A182 A183 A1831 A1832 A1839 A184 A185 A1850 A1851 A1852 A1853 A1854 A1859 A186 A187 A188 A1881 A1882 A1883 A1884 A1885 A1889 A19 A190 A191 A192 A198 A199 A20-A28 A20 A200 A201 A202 A203 A207 A208 A209 A21 A210 A211 A212 A213 A217 A218 A219 A22 A220 A221 A222 A227 A228 A229 A23 A230 A231 A232 A233 A238 A239 A

In [12]:
icd10_umls_pipeline.model.stages

[DocumentAssembler_25b094820972,
 REGEX_TOKENIZER_49a40e5cb790,
 CHUNKER-MAPPER_fc0d256341f7]

In [16]:
umls_code_list = icd10_umls_pipeline.annotate(my_string_clean)
pickle.dump(umls_code_list, open("umls_code_list.pkl", "wb"))

In [ ]:
icd10_umls_pipeline.annotate("M8950 R822 R0901")

{'document': ['M8950 R822 R0901'],
 'icd10cm_code': ['M8950', 'R822', 'R0901'],
 'umls_code': ['C4721411', 'C0159076', 'C0004044']}

|**ICD10CM Code** | **ICD10CM Details** | **UMLS Code** | **UMLS Details** |
| ---------- | -----------:| ---------- | -----------:|
| M8950 |  Osteolysis, unspecified site | C4721411 | osteolysis |
| R822 | Biliuria | C0159076 | Biliuria |
| R0901 | Asphyxia | C0004044 | Asphyxia |



## SNOMED to UMLS Code Mapping

This pretrained pipeline maps SNOMED codes to UMLS codes without using any text data. You’ll just feed white space delimited SNOMED codes and it will return the corresponding UMLS codes as a list. If there is no mapping, the original code is returned with no mapping.

In [ ]:
snomed_umls_pipeline = PretrainedPipeline( "snomed_umls_mapping","en","clinical/models")

snomed_umls_mapping download started this may take some time.
Approx size to download 4.9 MB
[OK!]


In [ ]:
snomed_umls_pipeline.model.stages

[DocumentAssembler_9e30bb9a34d6,
 REGEX_TOKENIZER_59c5ae0eef37,
 CHUNKER-MAPPER_6ecd1e714646]

In [ ]:
snomed_umls_pipeline.annotate('733187009 449433008 51264003')

{'document': ['733187009 449433008 51264003'],
 'snomed_code': ['733187009', '449433008', '51264003'],
 'umls_code': ['C4546029', 'C3164619', 'C0271267']}

|**SNOMED Code** | **SNOMED Details** | **UMLS Code** | **UMLS Details** |
| ---------- | -----------:| ---------- | -----------:|
| 733187009 | osteolysis following surgical procedure on skeletal system | C4546029 | osteolysis following surgical procedure on skeletal system |
| 449433008 | Diffuse stenosis of left pulmonary artery | C3164619 | diffuse stenosis of left pulmonary artery |
| 51264003 | Limbal AND/OR corneal involvement in vernal conjunctivitis | C0271267 | limbal and/or corneal involvement in vernal conjunctivitis |




## RXNORM to UMLS Code Mapping

This pretrained pipeline maps RxNorm codes to UMLS codes without using any text data. You’ll just feed white space-delimited RxNorm codes and it will return the corresponding UMLS codes as a list. If there is no mapping, the original code is returned with no mapping.

In [ ]:
rxnorm_umls_pipeline = PretrainedPipeline( "rxnorm_umls_mapping","en","clinical/models")

rxnorm_umls_mapping download started this may take some time.
Approx size to download 1.8 MB
[OK!]


In [ ]:
rxnorm_umls_pipeline.model.stages

[DocumentAssembler_4e185a5b8ed5,
 REGEX_TOKENIZER_fa6b16a107b9,
 CHUNKER-MAPPER_5ac1d410c34d]

In [ ]:
rxnorm_umls_pipeline.annotate("1161611 315677 343663")

{'document': ['1161611 315677 343663'],
 'rxnorm_code': ['1161611', '315677', '343663'],
 'umls_code': ['C3215948', 'C0984912', 'C1146501']}

|**RxNorm Code** | **RxNorm Details** | **UMLS Code** | **UMLS Details** |
| ---------- | -----------:| ---------- | -----------:|
| 1161611 |  metformin Pill | C3215948 | metformin pill |
| 315677 | cimetidine 100 mg | C0984912 | cimetidine 100 mg |
| 343663 | insulin lispro 50 UNT/ML | C1146501 | insulin lispro 50 unt/ml |

## MESH to UMLS Code Mapping

This pretrained pipeline maps MeSH codes to UMLS codes without using any text data. You’ll just feed white space delimited MeSH codes and it will return the corresponding UMLS codes as a list. If there is no mapping, the original code is returned with no mapping.

In [ ]:
mesh_umls_pipeline = PretrainedPipeline( "mesh_umls_mapping","en","clinical/models")

mesh_umls_mapping download started this may take some time.
Approx size to download 3.7 MB
[OK!]


In [ ]:
mesh_umls_pipeline.model.stages

[DocumentAssembler_7610a83bd4ea,
 REGEX_TOKENIZER_8b771a14973c,
 CHUNKER-MAPPER_4039b8f2b99c]

In [ ]:
mesh_umls_pipeline.annotate("C028491 D019326 C579867")

{'document': ['C028491 D019326 C579867'],
 'mesh_code': ['C028491', 'D019326', 'C579867'],
 'umls_code': ['C0043904', 'C0045010', 'C3696376']}

|**MeSH Code** | **MeSH Details** | **UMLS Code** | **UMLS Details** |
|-| -| - | -|
| C028491 |  1,3-butylene glycol | C0043904 | 1,3-butylene glycol |
| D019326 | 17-alpha-Hydroxyprogesterone | C0045010 | 17-alpha-hydroxyprogesterone |
| C579867 | 3-Methylglutaconic Aciduria | C3696376 | 3-Methylglutaconic Aciduria |

## RXNORM to MESH Code Mapping

This pretrained pipeline maps RxNorm codes to MeSH codes without using any text data. You’ll just feed white space-delimited RxNorm codes and it will return the corresponding MeSH codes as a list. If there is no mapping, the original code is returned with no mapping.

In [ ]:
rxnorm_mesh_pipeline = PretrainedPipeline( "rxnorm_mesh_mapping","en","clinical/models")

rxnorm_mesh_mapping download started this may take some time.
Approx size to download 101.1 KB
[OK!]


In [ ]:
rxnorm_mesh_pipeline.model.stages

[DocumentAssembler_d554433bf767,
 REGEX_TOKENIZER_e88353b947c6,
 LEMMATIZER_568c2c2ed9f2,
 Finisher_9aef0b33bc5c]

In [ ]:
rxnorm_mesh_pipeline.annotate("1191 6809 47613")

{'rxnorm': ['1191', '6809', '47613'],
 'mesh': ['D001241', 'D008687', 'D019355']}

|**RxNorm** | **Details** |
| ---------- | -----------:|
| 1191 |  aspirin |
| 6809 | metformin |
| 47613 | calcium citrate |

| **MeSH** | **Details** |
| ---------- | -----------:|
| D001241 | Aspirin |
| D008687 | Metformin |
| D019355 | Calcium Citrate |

## RXNORM to NDC Code Mapping

This pretrained pipeline maps RxNorm codes to NDC codes without using any text data. You’ll just feed white space-delimited RxNorm codes and it will return the corresponding NDC codes as a list.

In [ ]:
rxnorm_ndc_pipeline = PretrainedPipeline("rxnorm_ndc_mapping","en","clinical/models")

rxnorm_ndc_mapping download started this may take some time.
Approx size to download 3.9 MB
[OK!]


In [ ]:
rxnorm_ndc_pipeline.model.stages

[DocumentAssembler_f92d7dac3ac1,
 REGEX_TOKENIZER_4be4a602c96a,
 CHUNKER-MAPPER_2d7b0e176787,
 CHUNKER-MAPPER_2d7b0e176787]

In [ ]:
rxnorm_ndc_pipeline.annotate("1191 6809 47613")

{'document': ['1191 6809 47613'],
 'rxnorm_code': ['1191', '6809', '47613'],
 'package_ndc': ['62991-1176-06', '38779-2126-04', '00178-0796-30'],
 'product_ndc': ['62991-1176', '38779-2126', '00178-0796']}

|**RxNorm** | **Details** |
| ---------- | -----------:|
| 1191 |  aspirin |
| 6809 | metformin |
| 47613 | calcium citrate |